hyperion-shared-preferences/src/main/java/com/willowtreeapps/hyperion/sharedpreferences/list/SharedPreferencesListAdapter.java

```diff
@@ -0,0 +1,38 @@
+package com.willowtreeapps.hyperion.sharedpreferences.list;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.willowtreeapps.hyperion.sharedpreferences.R;
+
+import java.util.List;
+
+public class SharedPreferencesListAdapter extends RecyclerView.Adapter<SharedPreferencesFileViewHolder> {
+
+    private final List<String> preferences;
+
+    public SharedPreferencesListAdapter(List<String> preferences) {
+        this.preferences = preferences;
```

Comment:
I know at this point because of `Arrays.toList` we do in fact have an ArrayList style implementation however internally the adapter requires the list to be an ArrayList for performance reasons. The incoming preference list should be used to create an explicit ArrayList instance.

```java
this.preferences = new ArrayList(preferences);
```